app/controllers/api_controller/groups.rb

```diff
@@ -20,10 +20,15 @@ def edit_resource_groups(type, id, data)
       parse_set_tenant(data)
       group = resource_search(id, type, collection_class(:groups))
       parse_set_filters(data, :entitlement_id => group.entitlement.try(:id))
-      raise BadRequestError, "Cannot edit a read-only group" if group.read_only
+      raise Forbidden, "Cannot edit a read-only group" if group.read_only
       edit_resource(type, id, data)
     end
 
+    def delete_resource_groups(type, id, data = {})
+      raise Forbidden, "Cannot remove a read-only group" if MiqGroup.find(id).read_only?
```

Comment:
minor, Cannot delete a .... instead of remove, to be consistent with other messages.